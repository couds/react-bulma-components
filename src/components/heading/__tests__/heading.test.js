import React from 'react';
import { render } from 'react-testing-library';
import { Heading } from '..';

describe('Heading component', () => {
  it('should render', () => {
    const { asFragment } = render(<Heading />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Heading className="other-class this-is-a-test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(<Heading style={{ height: 250 }} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be a subtitle with size rendered as P', () => {
    const { asFragment } = render(
      <Heading size={3} subtitle renderAs="p">
        Subtitle
      </Heading>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not be spaced because is subtitle', () => {
    const { asFragment } = render(
      <Heading spaced subtitle renderAs="p">
        Subtitle
      </Heading>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be spaced', () => {
    const { asFragment } = render(
      <Heading spaced renderAs="p">
        Subtitle
      </Heading>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
