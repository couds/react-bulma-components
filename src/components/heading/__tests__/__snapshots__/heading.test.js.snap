// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should be a subtitle with size rendered as P 1`] = `
<DocumentFragment>
  <p
    class="subtitle is-3"
  >
    Subtitle
  </p>
</DocumentFragment>
`;

exports[`Heading component should be spaced 1`] = `
<DocumentFragment>
  <p
    class="title is-spaced"
  >
    Subtitle
  </p>
</DocumentFragment>
`;

exports[`Heading component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <h1
    class="other-class this-is-a-test title"
  />
</DocumentFragment>
`;

exports[`Heading component should not be spaced because is subtitle 1`] = `
<DocumentFragment>
  <p
    class="subtitle"
  >
    Subtitle
  </p>
</DocumentFragment>
`;

exports[`Heading component should render 1`] = `
<DocumentFragment>
  <h1
    class="title"
  />
</DocumentFragment>
`;

exports[`Heading component should use inline styles 1`] = `
<DocumentFragment>
  <h1
    class="title"
    style="height: 250px;"
  />
</DocumentFragment>
`;
